### DISCUSSION TITLE
Home position during heading

### DISCUSSION THREAD

**Discussion by stdakov** [mechano/Kingroon_KP3S_Pro_Marlin#5] [General]
During the heading the nozzle and bed the home position is in the middle of the bed and Y 0 and Z is almost 0.
![image](https://github.com/mechano/Kingroon_KP3S_Pro_Marlin_Firmware/assets/13567110/fd1e17ef-582a-4781-bfe6-c3fca8adbe2c)
Can you change the position to X 0 and Y 0? Or can you increase the Z, because when the nuzzle is on the bed during heading it causes clogs? 
Like this home position 
![image](https://github.com/mechano/Kingroon_KP3S_Pro_Marlin_Firmware/assets/13567110/ab3c390e-a4dd-4150-a4d9-aebbf6c46bc7)

**Comment by mechano** [comment]
Sorry but no one had this behavior with my firmware and KP3S Pro/S1.


Il giorno mer 25 ott 2023 alle ore 08:44 Stanislav Dakov <
***@***.***> ha scritto:

> During the heading the nozzle and bed the home position is in the middle
> of the bed and Y 0 and Z is almost 0.
> [image: image]
> <https://user-images.githubusercontent.com/13567110/277896064-fd1e17ef-582a-4781-bfe6-c3fca8adbe2c.png>
> Can you change the position to X 0 and Y 0? Or can you increase the Z,
> because when the nuzzle is on the bed during heading it causes clogs?
> Like this home position
> [image: image]
> <https://user-images.githubusercontent.com/13567110/277896531-ab3c390e-a4dd-4150-a4d9-aebbf6c46bc7.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mechano/Kingroon_KP3S_Pro_Marlin_Firmware/discussions/5>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAV5ZA7UJHXDKWFVTKKJJJLYBCYMPAVCNFSM6AAAAAA6O2U7ESVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZVG43TKMBRGQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>


-- 
Roberto Giuffrè
tel. +39-328-9140378
email: ***@***.***
pec: ***@***.***

**Reply by stdakov** [reply]
hmm that's odd, where is the nozzle positioned on your printer during heating before starting printing?

**Comment by mechano** [comment]
At first it homes all axis, after it goes to 0,0,10.75 (XYZ) and waits for
heating. On another printer KP5L I saw strange behavior with 0 only on X
after homing.  Same firmware, same configuration file.


Il giorno mer 25 ott 2023 alle ore 09:18 Stanislav Dakov <
***@***.***> ha scritto:

> hmm that's odd, where is the nozzle positioned on your printer during
> heating before starting printing?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mechano/Kingroon_KP3S_Pro_Marlin_Firmware/discussions/5#discussioncomment-7377914>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAV5ZAZCCONLMF6IP4EZYOLYBC4NJAVCNFSM6AAAAAA6O2U7ESVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TGNZXHEYTI>
> .
> You are receiving this because you commented.Message ID:
> <mechano/Kingroon_KP3S_Pro_Marlin_Firmware/repo-discussions/5/comments/7377914
> @github.com>
>


-- 
Roberto Giuffrè
tel. +39-328-9140378
email: ***@***.***
pec: ***@***.***

**Reply by stdakov** [reply]
ok I think I found the issue. It was in the slicer starting G codes (G1 Y1.0 Z0.3 F1000 ; move print head up): 
G90 ; use absolute coordinates
M83 ; extruder relative mode
M204 S[machine_max_acceleration_extruding] T[machine_max_acceleration_retracting]
M104 S[first_layer_temperature] ; set extruder temp
M140 S[first_layer_bed_temperature] ; set bed temp
G28 ; home all
G29; auto bed leveling
**G1 Y1.0 Z0.3 F1000 ; move print head up**
M190 S[first_layer_bed_temperature] ; wait for bed temp
M109 S[first_layer_temperature] ; wait for extruder temp
G92 E0.0
; initial load
 M117 Purge extruder
 G1 X2 Y20 Z0.3 F5000.0 ; move to start-line position
 G1 X2 Y175.0 Z0.3 F1500.0 E15 ; draw 1st line
 G1 X2 Y175.0 Z0.4 F5000.0 ; move to side a little
 G1 X2 Y20 Z0.4 F1500.0 E30 ; draw 2nd line
 G92 E0 ; reset extruder
 G1 Z1.0 F3000 ; move z up little to prevent scratching of surface

**Comment by mechano** [comment]
Which slicer?
I use Orca creating a new Marlin printer and configuring every aspect of
this printer: bed size, acceleration, speed, retraction, retraction speed,
nozzle size, etc.

It's a Marlin 2 firmware now and it can be different from a Marlin Legacy
profile.

Il giorno mer 25 ott 2023 alle ore 09:51 Stanislav Dakov <
***@***.***> ha scritto:

> ok I think I found the issue. It was in the slicer starting G codes (G1
> Y1.0 Z0.3 F1000 ; move print head up):
> G90 ; use absolute coordinates
> M83 ; extruder relative mode
> M204 S[machine_max_acceleration_extruding]
> T[machine_max_acceleration_retracting]
> M104 S[first_layer_temperature] ; set extruder temp
> M140 S[first_layer_bed_temperature] ; set bed temp
> G28 ; home all
> G29; auto bed leveling
> *G1 Y1.0 Z0.3 F1000 ; move print head up*
> M190 S[first_layer_bed_temperature] ; wait for bed temp
> M109 S[first_layer_temperature] ; wait for extruder temp
> G92 E0.0
> ; initial load
> M117 Purge extruder
> G1 X2 Y20 Z0.3 F5000.0 ; move to start-line position
> G1 X2 Y175.0 Z0.3 F1500.0 E15 ; draw 1st line
> G1 X2 Y175.0 Z0.4 F5000.0 ; move to side a little
> G1 X2 Y20 Z0.4 F1500.0 E30 ; draw 2nd line
> G92 E0 ; reset extruder
> G1 Z1.0 F3000 ; move z up little to prevent scratching of surface
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mechano/Kingroon_KP3S_Pro_Marlin_Firmware/discussions/5#discussioncomment-7378252>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAV5ZA3MZAKXXZIXH5WNNS3YBDAIXAVCNFSM6AAAAAA6O2U7ESVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TGNZYGI2TE>
> .
> You are receiving this because you commented.Message ID:
> <mechano/Kingroon_KP3S_Pro_Marlin_Firmware/repo-discussions/5/comments/7378252
> @github.com>
>


-- 
Roberto Giuffrè
tel. +39-328-9140378
email: ***@***.***
pec: ***@***.***

**Reply by stdakov** [reply]
Did you just use acceleration, speed, retraction, retraction speed,
nozzle size, etc. from kingroon profile ?

**Reply by mechano** [reply]
That was the base taken from Cura, after activate Input Shaper I moved to 2000 and tried also 3000mm/s² accel and 150-200mm/s printing speed. 
KP3S Pro S1 has a smoother fast printing than my Ender3 V2 Neo modified with X linear rails and Y rods, and better than KP5L with it's SGR15 rails. 

Titan has a base of 1.4mm and 40mm/s.

**Reply by stdakov** [reply]
Can you export the orca profile to test it?

**Reply by mechano** [reply]
I attach .3mf files of the ringing tower to tune input shaping.

[ringing_tower_input_shaper_KP3SPro_KP5L.zip](https://github.com/mechano/Kingroon_KP3S_Pro_Marlin_Firmware/files/13174112/ringing_tower_input_shaper_KP3SPro_KP5L.zip)
